test/src/org/exist/xmldb/XmldbURITest.java

```diff
@@ -323,39 +326,30 @@ public void xmldbURIConstructor15() throws URISyntaxException {
     }
 
     @Test
-    public void xmldbURIFaultyConstructor1() {
-        boolean exceptionThrown = false;
-        try{
-            @SuppressWarnings("unused")
-			XmldbURI xmldbURI = XmldbURI.xmldbUriFor("exist:///db");
-        } catch (URISyntaxException e) {
-            exceptionThrown = true;
-        }
-        assertTrue(exceptionThrown);
-        exceptionThrown = false;
-        try{
-            @SuppressWarnings("unused")
-            XmldbURI xmldbURI = XmldbURI.create("exist:///db");
-        } catch (Exception e) {
-            exceptionThrown = true;
-        }
-        assertTrue(exceptionThrown);
+    public void xmldbURIConstructor16() throws URISyntaxException {
+        XmldbURI xmldbURI = XmldbURI.xmldbUriFor("xmldb:///db/aa/bb/ccc");
+        assertEquals("xmldb", xmldbURI.getInstanceName());
+        assertNull(xmldbURI.getHost());
+        assertEquals(-1, xmldbURI.getPort());
+        assertNull(xmldbURI.getContext());
+        assertEquals("/db/aa/bb/ccc", xmldbURI.getCollectionPath());
+        assertEquals(XmldbURI.API_LOCAL, xmldbURI.getApiName());
     }
 
     @Test
-    public void xmldbURIFaultyConstructor2() {
+    public void xmldbURIFaultyConstructor1() {
         boolean exceptionThrown = false;
         try{
             @SuppressWarnings("unused")
-            XmldbURI xmldbURI = XmldbURI.xmldbUriFor("xmldb:///db");
+			XmldbURI xmldbURI = XmldbURI.xmldbUriFor("exist:///db");
```

Comment:
Some url resolvers having problems of pattern `scheme:[//[user:password@]host[:port]][/]path[?query][#fragment]` where `scheme` usually do not expect to have `:`.

Also, it's much shorter and possible to adopt to use as reference to "embedded" db (any, not just eXist-db).